[Merged by Bors] - feat(tactic/noncomm_ring): add noncomm_ring tactic

ocfnash opened the pull request:
Fixes https://github.com/leanprover-community/mathlib/issues/2727

Co-authored-by: Scott Morrison <scott.morrison@gmail.com>

Commit:
feat(tactic/noncomm_ring): add noncomm_ring tactic

Fixes https://github.com/leanprover-community/mathlib/issues/2727

ocfnash commented on src/algebra/classical_lie_algebras.lean:
```diff
@@ -84,7 +84,7 @@ begin
   intros c,
   have c' : A.val ⬝ B.val = B.val ⬝ A.val := by { rw [←sub_eq_zero, ←sl_bracket, c.abelian], refl, },
-  have : 1 = 0 := by simpa [matrix.mul_val, hij] using (congr_fun (congr_fun c' i) i),
+  have : (1 : R) = 0 := by simpa [matrix.mul_val, hij] using (congr_fun (congr_fun c' i) i),
```
This is an unrelated proposed change due to an observation of @kckennylau https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/a.20puzzle

ocfnash commented on src/tactic/noncomm_ring.lean:
````diff
@@ -0,0 +1,32 @@
+by noncomm_ring
+```
+-/
+add_tactic_doc
````
This is the first time I've attempted to tinker with tactics and I [do not fully understand](https://i.kym-cdn.com/entries/icons/original/000/008/342/ihave.jpg) the implications of what I pass to `add_tactic_doc` so I would be especially grateful for double checking here.

robertylewis replied:
I would move the doc string you've written here to `noncomm_ring` above. That doc string will appear as the entry for `noncomm_ring` at https://leanprover-community.github.io/mathlib_docs/tactics.html . Check that the tags you chose cohere with the existing ones there (in the drop down menu on the left). The `noncommutative` tag is unnecessary.

ocfnash replied:
Thanks, this is very helpful. I have made some adjustments based on these remarks.

Commit:
Apply suggestions of Rob Lewis

Also adding a few more tests

Commit:
Apply (part of) Scott Morrison's suggestions.

This will cause some tests to fail for now.

Commit:
Restore `mul_two`, `two_mul`

I still trust Scott Morrison that we should prefer to work via gsmul
but brief experiments suggest that this is not totally trivial. I
am thus reinstating the `mul_two` and `two_mul` so that tests pass,
pending further improvements.

kim-em committed:
better handling of scalars

kim-em committed:
move lemmas

kim-em committed:
oops, remove lemmas

kim-em committed:
comments

kim-em committed:
normalise grammar of comments

bryangingechen reviewed (commented):
Please add `import tactic.noncomm_ring` to `src/tactic/default.lean`.

Commit:
Add `noncomm_ring` to default tactic import.

Commit:
Tweak local notation priority in tests to obviate need for parentheses.
